x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -99,42 +96,72 @@ interface ExecuteActionHostResponseOutputProps {
 }
 
 export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
-  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize }) => {
+  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize = 'xs' }) => {
+    const { canWriteExecuteOperations } = useUserPrivileges().endpointPrivileges;
+
+    const [executeOutputContent, setExecuteOutputContent] = useState<
+      undefined | ResponseActionExecuteOutputContent
+    >(undefined);
+
     const outputContent = useMemo(
       () =>
         action.outputs &&
         action.outputs[agentId] &&
         (action.outputs[agentId].content as ResponseActionExecuteOutputContent),
-      [agentId, action]
+      [action.outputs, agentId]
     );
 
+    const { data: actionDetails, isFetching } = useGetActionDetails(action.id, {
+      enabled: canWriteExecuteOperations && !outputContent,
+    });
```

Comment:
@paul-tavares so after trying out the new changes for the action list API (`withOutputs`), I realized using the action details API was the best approach to fetch `execute` outputs on tray open and hence we're using the action details hook here. 

So now we use the action details that are fetched by the action list API in the tray details. This works for showing the output in console results for the `execute` action. However, on the actions log page and flyout, we fetch the details of the action using the action details API.



Let me know if we should revert back the action list API changes If we go with this approach.

Reply:
If we are not going to use the changes you made to the action list api, then yes, I agree - lets revert it.

Re: the action list

Just make sure that `refresh` works correctly on the action history log UI - especially when drawers are opened - to ensure that a refresh also triggers a refresh of the details api for the opened drawers.

btw: why are you checking `canWriteExecuteOperations` in the code above for `enable`? I don't think you need that.

Reply:
7a1f87b1bee40512a7d11b44ad16cd3d781748c6

Reply:
@paul-tavares So I thought about it a bit and looks like the best way to refresh data for open trays when refresh is triggered, is if we have an API that takes a list of action Ids and returns action details. So similar to the current action details API but one that accepts multiple action ids. 


We could do that also with the recent change to action list API that we added (`withOutputs`), but I don't find that good design as  we might end up with either:

1. Two sets of API requests for different UX (one for the individual opening of trays using action details API, the other for already open trays and triggering a refresh with page reload or auto-refresh/refresh button using action list API `withOutputs`) or
2. Using action list API for both individual details as well as details already open trays. In the latter case that would entail re-fetching the entire action list every time we have open trays.

Reply:
We talked offline and decided to try and use the action list API `withOutputs` for both individual and set of action outputs. Will create a new PR for if it works.